inc/scroom/pipettelayeroperations.hh

```diff
@@ -0,0 +1,47 @@
+/*
+ * Scroom - Generic viewer for 2D data
+ * Copyright (C) 2009-2020 Kees-Jan Dijkzeul
+ *
+ * SPDX-License-Identifier: LGPL-2.1
+ */
+
+#pragma once
+
+#include <string>
+#include <scroom/rectangle.hh>
+#include <scroom/tile.hh>
+
+/**
+ * This string defines the order in which color channels are stored in the map
+ * and eventually in what order the channels will be displayed to the user.
+ */
+const std::string COLOR_CHANNELS = "CMYK" "RGB" "ADEFHIJLNOPQSTUVWXZ";
+
+/**
+ * This struct is just here to compare two keys according to the order defined above
+ */
+struct ColorComparator : public std::binary_function<std::string, std::string, bool> {
```

Comment:
This breaks horribly when colors become more than one letter, as they will be for the "light" variants of CMYK

Reply:
Light variants of CMYK channels would be sorted to the end right now. However, we can instead compare character by character in a lexicographic fashion, with our custom alphabet. CMYK does not support light variants right now.

Reply:
Well, no, it is far worse than that. Assuming "LC", "LM" and "LY", your comparator will find none of them in the reference string. That causes `find()` to return -1. As a result, they are all considered equal. So the last one added wins, and it will be sorted before all others

I once again recommend you rethink your chosen datastructure.

(I also recommend writing some tests - then you'd be able to detect these kind of bugs much sooner, instead of relying on me to spot them)

Reply:
Hmmm. I am not really in the position to make a good argument for maps here now. Using `  typedef std::vector<std::pair<std::string, double>> PipetteColor` instead of the current `map` will get rid of the ordering issue and the changes required for this are fairly minimal (from what I have seen it's mainly the `sumPipetteColors()`) . The ordering will then be the same order as the vector (or could be sorted again). This would solve the problem i believe

Reply:
A vector of pairs would be much better, indeed. And as you say, probably only the sum and divide functions will be affected. Well, and of course the method in pipette plugin that shows the results